src/Microsoft.SqlTools.ServiceLayer/QueryExecution/ResultSet.cs

```diff
@@ -301,10 +381,10 @@ protected virtual void Dispose(bool disposing)
 
             IsBeingDisposed = true;
             // Check if saveTasks are running for this ResultSet
-            if (!saveTasks.IsEmpty)
+            if (!SaveTasks.IsEmpty)
             {
                 // Wait for tasks to finish before disposing ResultSet
-                Task.WhenAll(saveTasks.Values.ToArray()).ContinueWith((antecedent) =>
+                Task.WhenAll(SaveTasks.Values.ToArray()).ContinueWith((antecedent) =>
```

Comment:
>Task.WhenAll(SaveTasks.Values.ToArray()) [](start = 16, length = 40)

is this going to prevent the service host from exiting if a file save is in progress?  if so, is that the intended and reviewed design...saving a file can take several minutes, for example.